src/combinatorics/basic.lean

```diff
@@ -0,0 +1,61 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+
+import data.finset
+import data.fintype.basic
+
+open finset
+
+variable {α : Type*}
+variable {r : ℕ}
+
+/-!
+Basic definitions for finite sets which are useful for combinatorics.
+We define antichains, and a proposition asserting that a set is a set of r-sets.
+-/
+
+/--
+A family of sets is an antichain if no set is a subset of another. For example,
+{{1}, {4,6,7}, {2,4,5,6}} is an antichain.
+-/
+def antichain (𝒜 : finset (finset α)) : Prop := ∀ A ∈ 𝒜, ∀ B ∈ 𝒜, A ≠ B → ¬(A ⊆ B)
+
+/-- `all_sized A r` states that every set in A has size r. -/
+@[reducible]
+def all_sized (A : finset (finset α)) (r : ℕ) : Prop := ∀ x ∈ A, card x = r
+
+/--
+All sets in the union have size `r` iff both sets individually have this
+property.
+-/
+lemma union_layer [decidable_eq α] {A B : finset (finset α)} :
```

Comment:
There should be `all_sized.mono : all_sized A r → B ≤ A → all_sized B r` and `all_sized.union : all_sized A r → all_sized B r → all_sized (A ∪ B) r`